core/src/main/resources/org/apache/spark/ui/static/taskspages.js

```diff
@@ -0,0 +1,474 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+$(document).ajaxStop($.unblockUI);
+$(document).ajaxStart(function () {
+    $.blockUI({message: '<h3>Loading Tasks Page...</h3>'});
+});
+
+$.extend( $.fn.dataTable.ext.type.order, {
+    "file-size-pre": ConvertDurationString,
+
+    "file-size-asc": function ( a, b ) {
+        a = ConvertDurationString( a );
+        b = ConvertDurationString( b );
+        return ((a < b) ? -1 : ((a > b) ? 1 : 0));
+    },
+
+    "file-size-desc": function ( a, b ) {
+        a = ConvertDurationString( a );
+        b = ConvertDurationString( b );
+        return ((a < b) ? 1 : ((a > b) ? -1 : 0));
+    }
+} );
+
+function createTemplateURI(appId) {
+    var words = document.baseURI.split('/');
+    var ind = words.indexOf("proxy");
+    if (ind > 0) {
+        var baseURI = words.slice(0, ind + 1).join('/') + '/' + appId + '/static/stagespage-template.html';
+        return baseURI;
+    }
+    ind = words.indexOf("history");
+    if(ind > 0) {
+        var baseURI = words.slice(0, ind).join('/') + '/static/stagespage-template.html';
+        return baseURI;
+    }
+    return location.origin + "/static/stagespage-template.html";
+}
+
+// This function will only parse the URL under certain formate
+// e.g. https://axonitered-jt1.red.ygrid.yahoo.com:50509/history/application_1502220952225_59143/stages/stage/?id=0&attempt=0
+function StageEndPoint(appId) {
+    var words = document.baseURI.split('/');
+    var words2 = document.baseURI.split('?');
+    var ind = words.indexOf("proxy");
+    if (ind > 0) {
+        var appId = words[ind + 1];
+        var stageIdLen = words2[1].indexOf('&');
+        var stageId = words2[1].substr(3, stageIdLen - 3);
+        var newBaseURI = words.slice(0, ind + 2).join('/');
+        return newBaseURI + "/api/v1/applications/" + appId + "/stages/" + stageId;
+    }
+    ind = words.indexOf("history");
+    if (ind > 0) {
+        var appId = words[ind + 1];
+        var attemptId = words[ind + 2];
+        var stageIdLen = words2[1].indexOf('&');
+        var stageId = words2[1].substr(3, stageIdLen - 3);
+        var newBaseURI = words.slice(0, ind).join('/');
+        if (isNaN(attemptId) || attemptId == "0") {
+            return newBaseURI + "/api/v1/applications/" + appId + "/stages/" + stageId;
+        } else {
+            return newBaseURI + "/api/v1/applications/" + appId + "/" + attemptId + "/stages/" + stageId;
+        }
+    }
+    var stageIdLen = words2[1].indexOf('&');
+    var stageId = words2[1].substr(3, stageIdLen - 3);
```

Comment:
iiuc this doesn't account for anchors in the url (`#tasks-section`), I'd look into some url parsing js functions libs, I've seen some good ones before.

Reply:
Also I'm not quite sure what the `3`s are doing here

Reply:
So the javascript substr function takes the first param as the starting index and the second param as the length of characters. And the split happens way before so it is independent of any anchors. I have tested it for many cases and it seems to look well, but if you are still unsure, I can look into better ways of parsing the url.